flink/flink-common/src/main/java/com/netease/arctic/flink/FlinkSchemaUtil.java

```diff
@@ -197,4 +245,122 @@ public static void addPrimaryKey(
                               "Arctic primary key should be declared in table")));
         });
   }
+  /**
+   * Generate table properties for watermark and computed columns from flink TableSchema.
+   *
+   * @param schema flink TableSchema.
+   * @return Table properties map.
+   */
+  public static Map<String, String> addSchemaProperties(TableSchema schema) {
```

Comment:
How about 'generateExtraOptionsFrom'?
```suggestion
  public static Map<String, String> generateExtraOptionsFrom(TableSchema schema) {
```